From 9ea6fde9ce330f6c0786377824714f9741155ee3 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Fri, 15 Jan 2016 15:18:54 -0600
Subject: [PATCH 1/3] fix sleep33xx.S: for thumb2

<zmatt> rcn-ee: btw, I got a small patch that makes 4.1-ti compile in thumb mode
again, you want it?  (this means TI still compiles in ARM mode?  why oh why?)
<zmatt> http://pastebin.com/3S3sy8U4

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 arch/arm/mach-omap2/sleep33xx.S | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-omap2/sleep33xx.S b/arch/arm/mach-omap2/sleep33xx.S
index 26dba59..3cfc1ce 100644
--- a/arch/arm/mach-omap2/sleep33xx.S
+++ b/arch/arm/mach-omap2/sleep33xx.S
@@ -36,7 +36,9 @@
 ENTRY(am33xx_do_wfi)
 	stmfd	sp!, {r4 - r11, lr}	@ save registers on stack
 
-	str	r0, wfi_flags
+ARM(	str	r0, wfi_flags )
+THUMB(	adr	r2, wfi_flags )
+THUMB(	str	r0, [r2] )
 
 	/* Only necessary if PER is losing context */
 	ldr	r2, wfi_flags
@@ -200,7 +202,9 @@ wait_emif_enable1:
 	cmp	r1, r2
 	bne	wait_emif_enable1
 
-	adr	sp, temp_stack
+ARM(	adr	sp, temp_stack )
+THUMB(	adr	r1, temp_stack )
+THUMB(	mov	sp, r1 )
 
 	ldr	r1, ti_emif_restore_context
 	blx	r1
-- 
2.7.0.rc3

